core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -494,4 +490,24 @@ object Endpoint {
    * Generically derive a very basic instance of [[Endpoint]] for a given type `A`.
    */
   def derive[A]: GenericDerivation[A] = new GenericDerivation[A]
+
+  implicit def applicativeEndpoint: Applicative[Endpoint] = new Applicative[Endpoint] {
+
+    override def pure[A](x: A): Endpoint[A] = new Endpoint[A] {
+      override def apply(input: Input): Result[A] = None
+    }
+
+    override def ap[A, B](ff: Endpoint[(A) => B])(fa: Endpoint[A]): Endpoint[B] = fa.ap(ff)
+
+    override def product[A, B](fa: Endpoint[A], fb: Endpoint[B]): Endpoint[(A, B)] = {
+      fa.ap(fb.map(b => (a: A) => a -> b))
+    }
+
+    override def map[A, B](fa: Endpoint[A])(f: (A) => B): Endpoint[B] = fa.map(f)
+  }
+
+
+  implicit def functorEndpoint: Functor[Endpoint] = new Functor[Endpoint] {
```

Comment:
I don't think we need this. `Applicative` extends `Functor` at the end of the day.